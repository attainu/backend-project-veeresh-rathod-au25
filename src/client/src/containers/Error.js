import React from "react";

import { Error } from "../components";

const ErrorContainer = (props) => {
  let msg = "Invalid session. Please login again!";
  return <Error message={msg} {...props} />;
};

export default ErrorContainer;