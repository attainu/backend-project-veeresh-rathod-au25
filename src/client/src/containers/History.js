import React from "react";
import PropTypes from "prop-types";

import { History } from "../components";

import * as Hocs from "../hocs";
import { getUserHistory } from "../actions";

export const HistoryContainer = (props) => {
  React.useEffect(() => {
    getUserHistory();
  }, []);

  return <History {...props} />;
};

HistoryContainer.propTypes = {
  users: PropTypes.array
};

const stateMapper = (state) => ({
  users: state.userHistory.payload || []
});

const HistoryWithAuth = Hocs.withAuth(HistoryContainer);
export default Hocs.withStore(stateMapper)(HistoryWithAuth);