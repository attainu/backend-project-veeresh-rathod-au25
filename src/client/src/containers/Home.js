import React from "react";
import PropTypes from "prop-types";

import { Home } from "../components";

import { withStore, withAuth } from "../hocs";
import { getActiveUsers } from "../actions";
import Socket from "../utils/socket";

export const HomeContainer = (props) => {
  React.useEffect(() => {
    if (!Socket.getSocket())
      Socket.create();
    getActiveUsers();
  }, []);

  return <Home {...props} />;
};

HomeContainer.propTypes = {
  activeUsers: PropTypes.object,
  loggedInUser: PropTypes.object
};

const stateMapper = (state) => ({
  activeUsers: Object.values(state.activeUsers.payload),
  loggedInUser: state.login.payload && state.login.payload.user
});

const HomeWithAuth = withAuth(HomeContainer);
export default withStore(stateMapper)(HomeWithAuth);