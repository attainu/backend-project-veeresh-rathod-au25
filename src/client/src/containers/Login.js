import React from "react";
import PropTypes from "prop-types";

import { Login } from "../components";
import { withStore } from "../hocs";
import ReactRouterDom from "react-router-dom";

export const LoginContainer = (props) => {
  const history = ReactRouterDom.useHistory();
  const { loggingIn, loginSuccess } = props;

  React.useEffect(() => {
    if (!loggingIn && loginSuccess) {
      history.push("/app/home");
    }
  }, [loggingIn, loginSuccess]);

  return <Login {...props} />;
};

LoginContainer.propTypes = {
  loggingIn: PropTypes.bool,
  loginSuccess: PropTypes.bool,
  loginFailure: PropTypes.bool,
  signingUp: PropTypes.bool,
  signUpSuccess: PropTypes.bool,
  loggingOut: PropTypes.bool,
  logoutSuccess: PropTypes.bool
};

const stateMapper = (state) => ({
  loggingIn: state.login.loading,
  loginSuccess: state.login.payload,
  loginFailure: state.login.error,
  signingUp: state.signUp.loading,
  signUpSuccess: state.signUp.payload,
  loggingOut: state.logout.loading,
  logoutSuccess: state.logout.payload,
});

export default withStore(stateMapper)(LoginContainer);
