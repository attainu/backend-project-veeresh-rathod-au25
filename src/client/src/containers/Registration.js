import React from "react";
import PropTypes from "prop-types";
import ReactRouterDom from "react-router-dom";

import { Registration } from "../components";
import { withStore } from "../hocs";
import { SIGN_UP_RESET } from "../utils/actions";

export const RegistrationContainer = (props) => {
  const history = ReactRouterDom.useHistory();
  const { signingUp, signUpSuccess, dispatch } = props;

  React.useEffect(() => {
    if (!signingUp && signUpSuccess) {
      history.push("/app");
      setTimeout(() => dispatch({ type: SIGN_UP_RESET }), 3000);
    }
  }, [signingUp, signUpSuccess]);

  return <Registration {...props} />
};

RegistrationContainer.propTypes = {
  signingUp: PropTypes.bool.isRequired,
  signUpSuccess: PropTypes.bool,
  signUpFailure: PropTypes.bool,
  dispatch: PropTypes.func.isRequired
};

const stateMapper = (state) => ({
  signingUp: state.signUp.loading,
  signUpSuccess: state.signUp.payload,
  signUpFailure: state.signUp.error,
});

export default withStore(stateMapper)(RegistrationContainer);