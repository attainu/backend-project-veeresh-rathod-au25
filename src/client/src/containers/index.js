export { default as Login } from "./Login";
export { default as Home } from "./Home";
export { default as History } from "./History";
export { default as Registration } from "./Registration";
export { default as Error } from "./Error";
