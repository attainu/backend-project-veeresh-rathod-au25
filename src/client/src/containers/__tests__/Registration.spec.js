jest.mock("react-router-dom");

import React from "react";
import Registration from "../../components/Registration";
import { RegistrationContainer } from "../../containers/Registration";
import { shallow } from "enzyme";
import ReactRouterDom from "react-router-dom";
ReactRouterDom.useHistory = jest.fn().mockReturnValue({
  push: jest.fn()
});

describe("CONTAINER: src/containers/Registration.js", () => {
  const defaultProps = {
    dispatch: () => { },
    signingUp: true,
    signUpSuccess: false
  };
  const render = (props) => shallow(<RegistrationContainer {...defaultProps} {...props} />);
  beforeEach(() => {
    React.useEffect = jest.fn().mockImplementation(f => f());
  });

  it("should match snapshot", () => {
    expect(render()).toMatchSnapshot();
  });

  it("should render Registration component", () => {
    expect(render().find(Registration)).toBeDefined();
  });

  it("should proceed to login page if user is registered", () => {
    render({ signingUp: false, signUpSuccess: true });
    expect(ReactRouterDom.useHistory().push).toHaveBeenCalled();
  });
});