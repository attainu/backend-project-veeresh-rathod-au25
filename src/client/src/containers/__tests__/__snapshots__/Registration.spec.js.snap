// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTAINER: src/containers/Registration.js should match snapshot 1`] = `ShallowWrapper {}`;
